### PR TITLE
Disable NNAPI support for dynamic input shape, add warning logs

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Disable NNAPI support for dynamic input shape, add warning logs

**Motivation and Context**
- Since NNAPI does not support hardware acceleration for graph with dynamic input shape (it will still run with worse perf than our CPU EP), disable it for now
- Add warning log for NNAPI/CoreML EP if the graph is partitioned in a way that NNAPI will take more than 1 partitions, for example a graph is partitioned as shown below, the device copies may totally offset the benefits of using hardware accelerations. Need to warn users this may impact perf.
```
CPU
|
NNAPI
|
CPU
|
NNAPI
```

### CODE REVIEW

File: onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc [hunk 1]
```diff
@@ -196,10 +196,19 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
...
+  auto num_of_partitions = result.size();
+  const auto summary_msg = MakeString(
+      "NnapiExecutionProvider::GetCapability,",
+      " number of partitions supported by NNAPI: ", num_of_partitions,
+      " number of nodes in the graph: ", graph_view.NumberOfNodes(),
+      " number of nodes supported by NNAPI: ", num_of_supported_nodes);
+
+  // If the graph is partitioned in multiple subgraphs, and this may impact performance,
+  // we want to give users a summary message at warning level.
+  if (num_of_partitions > 1)
```
Comment (gwang-msft) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc:208]: We can log warning message when num_of_partitions > certain number, 1 might not be the best choice here

File: onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc [hunk 2]
```diff
@@ -483,6 +483,27 @@ bool IsNodeSupportedInternal(const Node& node, const GraphViewer& graph_viewer,
...
 }
 
+bool IsInputSupported(const NodeArg& input, const std::string& parent_name) {
+  const auto& input_name = input.Name();
+  const auto* shape_proto = input.Shape();
+  // We do not support input with no shape
+  if (!shape_proto) {
+    LOGS_DEFAULT(VERBOSE) << "Input [" << input_name << "] of [" << parent_name
+                          << "] has not shape";
+    return false;
```
Comment (skottmckay) [inline onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc:493]: nit: has no shape

**Review by skottmckay — APPROVED** [review]
:shipit:

File: onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc (hunk 2 above)
Comment (gwang-msft) [reply onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc:493]: Will fix in my next other PR

**Pull request merged**